spinalcordtoolbox/vertebrae/detect_c2c3.py

```diff
@@ -121,7 +116,10 @@ def detect_c2c3(nii_im, nii_seg, contrast, nb_sag_avg=7.0, verbose=1):
     tmp_folder.cleanup()
 
     nii_c2c3.change_orientation(orientation_init)
-    return nii_c2c3
+    if verbose < 2:
```

Comment:
why not simply returning everything, then in `detect_c2c3_from_file` you do:
~~~
nii_c2c3, nii_midSlice, nii_mask, nii_after_postPro, nii_before_postPro = detect_c2c3(nii_im.copy(),
                                                                                                nii_seg,
                                                                                                contrast,
                                                                                                verbose=verbose)
if verbose > 1:
    # save debugging files
    path_temp = sct.tmp_create
    nii_midSlice.save(os.path.join(os.path.dirname(nii_im.absolutepath), os.path.join(path_temp, "tmp_midSlice.nii.gz")))
   # etc.
~~~

also, good habit to use `sct.temp_dir`, to avoid overwriting, in case use likes to call his files "tmp_midSlice.nii.gz"

Reply:
update: actually, it would be less confusing to save the debugging files inside the calling function (not in  detect_c2c3_from_file)

Reply:
Good point!
Done